File: services/notification-service/DTOs/NotificationDto.cs
using System.Text.Json.Serialization;

namespace notification_service.DTOs;

public class NotificationDto {
    [JsonPropertyName("id")] public long Id { get; set; }
    [JsonPropertyName("eventId")] public string EventId { get; set; } = null!;
    [JsonPropertyName("type")] public string Type { get; set; } = null!;
    [JsonPropertyName("title")] public string Title { get; set; } = null!;
    [JsonPropertyName("message")] public string Message { get; set; } = null!;
    [JsonPropertyName("actorUserId")] public string ActorUserId { get; set; } = null!;
    [JsonPropertyName("resourceType")] public string ResourceType { get; set; } = null!;
    [JsonPropertyName("resourceId")] public string ResourceId { get; set; } = null!;
    [JsonPropertyName("metadata")] public Dictionary<string, object>? Metadata { get; set; }
    [JsonPropertyName("createdAt")] public DateTime CreatedAt { get; set; }
    [JsonPropertyName("readAt")] public DateTime? ReadAt { get; set; }
}

File: client/src/app/shared/models/notification/notification.dto.ts
import { NotificationType } from "./notification-type.enum";

export interface NotificationDto {
  id: string;
  eventId: string;
  type: NotificationType;
  title: string;
  message: string;
  actorUserId: string;
  resourceType: string;
  resourceId: string;
  metadata: Record<string, any>;
  createdAt: string;
  readAt: string | null;
  read?: boolean;
}
